### PR TITLE
feat: additional security considerations

### PR DESCRIPTION
**Pull request by peppelinux**
# Security Considerations

- feat: Minimum length of the salt
- feat: weak hash algorithms to not use

### CODE REVIEW

**Comment by peppelinux** [comment]
mh ... we already have `MUST contain at least 128 bits of pseudorandom data` in Section [SD_JWT Claims](https://github.com/oauthstuff/draft-selective-disclosure-jwt/blob/master/main.md#sd-jwt-claims)

File: main.md [hunk 1]
```diff
@@ -682,12 +682,20 @@ adhere to this principle. As such, the salt has to be
...
+ 
+
 ## Choice of a hash function
 
 For the security of this scheme, the hash function is required to have the following property.
 Given a claim value, a salt, and the resulting hash, it is hard to find a second salt value 
 so that HASH(salt | claim_value) equals the hash.
 
+Furthermore the hash algorithms MD2, MD4, MD5, RIPEMD-160, and SHA-1 
+revealed fundamental weaknesses and they should no longer be used.
```
Comment (danielfett) [inline main.md:697]:
```suggestion
revealed fundamental weaknesses and they MUST NOT be used.
```

File: main.md [hunk 2]
```diff
@@ -682,12 +682,20 @@ adhere to this principle. As such, the salt has to be
 created in such a manner that it is cryptographically random, long enough and has
 high entropy that it is not practical for the attacker to guess.
 
+## Minimum length of the salt
+
+The length of the randomly-generated portion of the salt shall be at least 128 bits.
```
Comment (danielfett) [inline main.md:687]:
```suggestion
The length of the randomly-generated portion of the salt MUST be at least 128 bits.
```

**Review by danielfett — CHANGES_REQUESTED** [review]

**Comment by peppelinux** [comment]
@danielfett I commited your suggestion

I'm not sure that the normative language could be used in the considerations but it's good anyway to me, it's intellegibile :)

**Review by Sakurann — APPROVED** [review]

**Pull request merged**